internal/handler/wallet/sign.go

```diff
@@ -73,40 +70,31 @@ func (s *walletSign) Handle(params operations.SignOperationParams) middleware.Re
 		return resp
 	}
 
-	_, signature, resp := sign(wlt, params)
-	if resp != nil {
-		return resp
+	signature, err := wlt.Sign(params.Body.Operation)
+	if err != nil {
+		return operations.NewSignInternalServerError().WithPayload(
+			&models.Error{
+				Code:    errorSignRead,
+				Message: "Error: while reading operation.",
+			})
 	}
 
-	return operations.NewSignOperationOK().WithPayload(
-		&models.Signature{
+	return operations.NewSignOK().WithPayload(
+		&models.SignResponse{
 			PublicKey:     wlt.GetPupKey(),
 			Signature:     signature,
 			CorrelationID: correlationId,
 		})
 }
 
-func sign(wlt *wallet.Wallet, params operations.SignOperationParams) ([]byte, []byte, middleware.Responder) {
-	pubKey := wlt.KeyPair.PublicKey
-	privKey := wlt.KeyPair.PrivateKey
-
-	digest, resp := digestOperationAndPubKey(params.Body.Operation, pubKey)
-	if resp != nil {
-		return nil, nil, resp
-	}
-
-	signature := ed25519.Sign(privKey, digest[:])
-	return pubKey, signature, nil
-}
-
-func handleWithCorrelationId(wlt *wallet.Wallet, params operations.SignOperationParams, gc gcache.Cache) (models.CorrelationID, middleware.Responder) {
+func handleWithCorrelationId(wlt *wallet.Wallet, params operations.SignParams, gc gcache.Cache) (models.CorrelationID, middleware.Responder) {
 	cacheKey := getCacheKey(params.Body.CorrelationID)
```

Comment:
I don't understand

Reply:
Sorry I mixed two things. The fact that getting information is fast. here using get make sens. But in go, usingt get is a bad habit when is doesn't add information (see https://go.dev/doc/effective_go#Getters). Here you should remove the get prefix because it's simply useless...